resources/services/ec2/instance_statuses.go

```diff
@@ -0,0 +1,163 @@
+				Description: "The system instance health or application instance health.",
+				Type:        schema.TypeJSON,
+				Resolver:    schema.PathResolver("SystemStatus.Details"),
+			},
+			{
+				Name:        "system_status_status",
+				Description: "The system status.",
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("SystemStatus.Status"),
+			},
+		},
+		Relations: []*schema.Table{
+			{
+				Name:        "aws_ec2_instance_status_events",
+				Description: "Any scheduled events associated with the instance.",
+				Resolver:    fetchEc2InstanceStatusEvents,
+				Options:     schema.TableCreationOptions{PrimaryKeys: []string{"instance_status_cq_id", "id"}},
+				Columns: []schema.Column{
+					{
+						Name:        "instance_status_cq_id",
+						Description: "Unique CloudQuery ID of aws_ec2_instance_statuses table (FK)",
+						Type:        schema.TypeUUID,
+						Resolver:    schema.ParentIdResolver,
+					},
+					{
+						Name:        "code",
+						Description: "The event code.",
+						Type:        schema.TypeString,
+					},
+					{
+						Name:        "description",
+						Description: "A description of the event.",
+						Type:        schema.TypeString,
+					},
+					{
+						Name:        "id",
+						Description: "The ID of the event.",
+						Type:        schema.TypeString,
+						Resolver:    schema.PathResolver("InstanceEventId"),
+					},
+					{
+						Name:        "not_after",
+						Description: "The latest scheduled end time for the event.",
+						Type:        schema.TypeTimestamp,
+					},
+					{
+						Name:        "not_before",
+						Description: "The earliest scheduled start time for the event.",
+						Type:        schema.TypeTimestamp,
+					},
+					{
+						Name:        "not_before_deadline",
+						Description: "The deadline for starting the event.",
+						Type:        schema.TypeTimestamp,
+					},
+				},
+			},
+		},
+	}
+}
+
+// ====================================================================================================================
+//                                               Table Resolver Functions
+// ====================================================================================================================
+
+func fetchEc2InstanceStatuses(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
+	var config ec2.DescribeInstanceStatusInput
+	c := meta.(*client.Client)
+	svc := c.Services().EC2
+	for {
+		output, err := svc.DescribeInstanceStatus(ctx, &config, func(options *ec2.Options) {
+			options.Region = c.Region
+		})
+		if err != nil {
+			return err
+		}
+		res <- output.InstanceStatuses
+		if aws.ToString(output.NextToken) == "" {
+			break
+		}
+		config.NextToken = output.NextToken
+	}
+	return nil
+}
+func fetchEc2InstanceStatusEvents(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
+	r := parent.Item.(types.InstanceStatus)
```

Comment:
I would check the casting was successful:

```suggestion
	r, ok := parent.Item.(types.InstanceStatus)
	if !ok {
		return fmt.Errorf("not ec2 instance status")
	}
	
```